File: programmers/위장/위장.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> res;
    for (auto cloth: clothes) res[cloth[1]]++;
    for (auto result: res) answer *= (result.second + 1);
    return answer - 1;
}

File: programmers/위장/위장.kt
import java.util.*
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1
        val closet = TreeMap<String, Int>()
        clothes.forEach { cloth -> closet[cloth[1]] = closet.getOrDefault(cloth[1], 0) + 1 }
        for (cloth in closet) answer *= (cloth.value + 1)
        return answer - 1
    }
}